File: backend/app/schemas/notification.py
import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class NotificationResponse(BaseModel):
    id: uuid.UUID
    channel: str
    event_type: str
    title: str
    body: Optional[str] = None
    is_read: bool
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/notification.ts
export type NotificationChannel = "in_app" | "email" | "sms" | "push";

export interface AppNotification {
  id: string;
  channel: NotificationChannel;
  event_type: string;
  title: string;
  body?: string | null;
  is_read: boolean;
  created_at: string;
}
